config/services/docker-swarm.xml

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<service>
+  <short>Docker swarm</short>
+  <description>Ports and protocol required by docker in swarm mode</description>
```

Comment:
This should be a one line description of Docker Swarm not a description of the service file.

Reply:
TCP port 2376 is not required by docker engines running in swarm mode. See https://docs.docker.com/engine/swarm/swarm-tutorial/#open-protocols-and-ports-between-the-hosts

Your link describes legacy standalone swarm used on docker prior the version 1.12. See https://docs.docker.com/swarm/provision-with-machine/

So if you want also open port 2376 needed for docker-machine and legacy swarm, I recommend have two firewalld services. One for "Docker legacy Swarm" also with port 2376 and one for "Docker integrated Swarm" without 2376 port.